### DISCUSSION TITLE
How to run a command line command from a nelua file

### DISCUSSION THREAD

**Discussion by ghost** [edubart/nelua-lang#142] [Q&A] [answered]
by "command line command" i mean commands like
"cd", "mkdir", "ls", "nelua.bat", etc

**Comment by stefanos82** [comment]
I'm on Linux and the following code works just fine for me:

```lua
require 'os'

os.execute('clear') -- clears terminal.

**Reply by ghost** [reply]
Works for me also, thanks

**Comment by edubart** [comment, marked as answer]
You can run command line commands like in Lua, using the `os` or `io` modules, you could check Lua documentation or tutorials on how to do that. You are probably looking into using `os.execute` in case you want to execute a command without capturing its output, or `io.popen` in case you want to capture the output of a command.

Here is a small example:
```lua
require 'os'
require 'io'

-- Run a command, the output will go to terminal `stdout`.
os.execute('ls')

-- Run a command, the output will go to string `out`.
local f = assert(io.popen('ls'))
local out = assert(f:read('a'))
f:close()
print(out)
```

**Reply by ghost** [reply]
Thanks, i just needed the one without the output for now, but the one which captures output is going to be helpful for the compile bot.

**Comment by ghost** [comment]
```lua
-- run.nelua
require 'os'
require 'io'

local output_file: filestream

local run = io.popen("nelua code.nelua")
local output = run:read('a')

output_file = io.open("output.txt", "w+")
output_file:write(output)
output_file:flush()

output_file:close()
run:close()
```
```lua
-- code.nelua
print("Hello World"
```

1. If i compile and run run.nelua, it would print in the console that there's an error in code.nelua even though i don't have any print statement in run.nelua.
2. the errors which are printed in the console are not in output.txt

So is there a way to capture the output even if the compilation of the nelua file fails?

**Reply by ghost** [reply]
Never Mind, I found a workaround

**Reply by edubart** [reply]
> Never Mind, I found a workaround

One way I workaround this is by executing a command while saving the output to a file like:
```lua
os.execute("command > command_stdout 2> command_stderr")
```
This will save stdout to `command_stdout` file and stderr to `command_stderr` file, you can then read it.

**Reply by ghost** [reply]
that's the exact workaround I'm using :sweat_smile: 
```
"nelua code.nelua > output.txt 2> error.txt"
```
